GVFS/GVFS.Common/Tracing/JsonTracer.cs

```diff
@@ -83,13 +90,22 @@ public bool HasLogFileEventListener
 
         public void AddEventListener(EventListener listener)
         {
+            if (Volatile.Read(ref this.isDisposed) == DISPOSED_TRUE)

```

Comment:
Are `AddEventListener` and `Dispose` ever called on different threads?

If not, I'm not sure that we need to make those methods thread safe.

It seems like in most places the pattern is:

```
using (JsonTracer tracer = new JsonTracer(...))
{
    // Add event listeners

    // Do stuff
}
```

Have you found spots in the code where we're not doing this?  If there are spots that are not thread safe, would it be simpler to update that code to be safe (rather than making `AddEventListener` and `Dispose` thread-safe)?

Reply:
> Have you found spots in the code where we're not doing this? If there are spots that are not thread safe, would it be simpler to update that code to be safe (rather than making AddEventListener and Dispose thread-safe)?

@wilbaker I'll just back out changes in `JsonTracer` regarding Add/Dispose.